xbmc/dbwrappers/dataset.cpp

```diff
@@ -420,21 +420,24 @@ const field_value Dataset::f_old(const char *f_name) {
   return fv;
 }
 
-int Dataset::str_compare(const char * s1, const char * s2) {
- 	std::string ts1 = s1;
- 	std::string ts2 = s2;
- 	std::string::const_iterator p = ts1.begin();
- 	std::string::const_iterator p2 = ts2.begin();
- 	while (p!=ts1.end() && p2 != ts2.end()) {
- 		if (toupper(*p)!=toupper(*p2))
- 			return (toupper(*p)<toupper(*p2)) ? -1 : 1;
- 		++p;
- 		++p2;
- 	}
- 	return (ts2.size() == ts1.size())? 0:
- 		(ts1.size()<ts2.size())? -1 : 1;
- }
+int Dataset::str_compare(const char* s1, const char* s2)
```

Comment:
What about using `StringUtils::CompareNoCase` instead?

Reply:
Top, will change that.